pkg/endpointmanager/manager.go

```diff
@@ -125,7 +125,7 @@ func New(logger *slog.Logger, registry *metrics.Registry, epSynchronizer Endpoin
 		policyUpdateCallbackDetails:  newPolicyUpdateCallbackDetails(),
 	}
 	mgr.deleteEndpoint = mgr.removeEndpoint
-	mgr.policyMapPressure = newPolicyMapPressure(logger, registry)
+	mgr.policyMapPressure = newPolicyMapPressure(logger, registry, config.BPFPolicyMapPressureMetricsThreshold)
```

Comment:
I don't think newPolicyMapPressure or the underlying "gauge with threshold" validate this input.  It's probably worth doing some diligence on whether the value is not less than zero (note: above 1 is actually fine since this metric can go above 100%).

Reply:
I think code wise this all makes sense, I do have one thing I'm wondering about.  There are many maps that that use the "pressure threshold" metrics (see: references to `WithPressureMetric`).   Although this change is useful for policy map thresholds, and that's probably the most commonly used map for this metric; I wonder if in the future we might run into someone wanting to have a similar configurable threshold for other maps that support the pressure metric.  In this case it would become unweildly to have a separate such metric flag for each map type.

I wonder if we may want to explore adding a generic flag that allows to configure thresholds per map type.